### PR TITLE
Update to latest core/m4

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This requires updating gotest to the latest testmodeler which uses a compatible version of core.
Added support for new SchemaType ArmId.
Added missing import for io in some cases.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@tadelesh can you please update this PR with a fix for the jest build failures?  I believe it has something to do with an improperly set `moduleDirectories` but I haven't been able to resolve it (I'm not familiar with jest).

**Comment by tadelesh** [comment]
@jhendrixMSFT I've temporally resolved the problem that could unblock your release. The latest version of testmodeler has some pack issue. I will work with @raych1 to resolve it in another PR.

**Comment by jhendrixMSFT** [comment]
@tadelesh thanks for the fast turnaround!

File: packages/autorest.go/test/autorest/mediatypesgroupwithnormailzedoperationname/zz_constants.go [hunk 1]
```diff
@@ -52,41 +52,41 @@ func PossibleContentType1Values() []ContentType1 {
 	}
 }
 
-// ContentType1AutoGenerated - Content type for upload
-type ContentType1AutoGenerated string
+// ContentType2 - Content type for upload
+type ContentType2 string
```
Comment (chlowell) [inline packages/autorest.go/test/autorest/mediatypesgroupwithnormailzedoperationname/zz_constants.go:56]: I'm curious whether we could see a change like this in a module we've shipped but I suppose if we did, we'd simply add a transform to keep the released name.

**Review by chlowell — APPROVED** [review]

File: packages/autorest.go/test/autorest/mediatypesgroupwithnormailzedoperationname/zz_constants.go (hunk 1 above)
Comment (jhendrixMSFT) [reply packages/autorest.go/test/autorest/mediatypesgroupwithnormailzedoperationname/zz_constants.go:56]: Fortunately, APIs that specify multiple content types are very rare so I don't expect this to be a problem in practice.  You're correct though that if this were to become a problem we can rename via a transform (which is unfortunate as the new names a better IMO).

**Comment by jhendrixMSFT** [comment]
@tadelesh terribly sorry but could you please update this again?  I lost the changes when I rebased and rerunning the tests deletes all of the content for armcompute.

**Comment by tadelesh** [comment]
> @tadelesh Chenjie Shi FTE terribly sorry but could you please update this again? I lost the changes when I rebased and rerunning the tests deletes all of the content for armcompute.

fixed.

**Review by tadelesh — APPROVED** [review]

**Pull request merged**